tests/foreman/api/test_contentviewversion.py

```diff
@@ -42,52 +41,42 @@ def test_delete_version(self):
 
         """
         key_content = read_data_file(ZOO_CUSTOM_GPG_KEY)
-        org_id = entities.Organization().create_json()['id']
-        gpgkey_id = entities.GPGKey(
+        org = entities.Organization().create()
+        gpgkey = entities.GPGKey(
             content=key_content,
-            organization=org_id
-        ).create_json()['id']
+            organization=org,
+        ).create()
         # Creates new product without selecting GPGkey
-        product_id = entities.Product(
-            organization=org_id
-        ).create_json()['id']
+        product = entities.Product(organization=org).create()
         # Creates new repository with GPGKey
         repo = entities.Repository(
+            gpg_key=gpgkey,
+            product=product,
             url=FAKE_1_YUM_REPO,
-            product=product_id,
-            gpg_key=gpgkey_id,
         ).create()
         # sync repository
         repo.sync()
         # Create content view
-        content_view = entities.ContentView(
-            organization=org_id
-        ).create()
+        content_view = entities.ContentView(organization=org).create()
         # Associate repository to new content view
-        client.put(
-            content_view.path(),
-            {u'repository_ids': [repo.id]},
-            auth=get_server_credentials(),
-            verify=False,
-        ).raise_for_status()
+        content_view.repository = [repo]
+        content_view = content_view.update(['repository'])
         # Publish content view
         content_view.publish()
+        content_view = content_view.read()
         # Get published content-view version id
-        cv = entities.ContentView(id=content_view.id).read_json()
-        self.assertEqual(len(cv['versions']), 1)
-        cv_version_id = cv['versions'][0]['id']
+        self.assertEqual(len(content_view.version), 1)
+        self.assertEqual(len(content_view.version[0].id), 1)
         # Get 'Library' life-cycle environment id
         response = client.get(
```

Comment:
Nope. The `read()` method is only good for reading a single entity. It's not good for performing searches. Searching is a huge topic that I haven't figured out how to tackle yet.